userapi/storage/postgres/registration_tokens_table.go

```diff
@@ -0,0 +1,250 @@
+	if value == 0 {
+		return nil
+	}
+	return value
+}
+
+func nullIfZeroInt32(value int32) interface{} {
+	if value == 0 {
+		return nil
+	}
+	return value
+}
+
+func (s *registrationTokenStatements) ListRegistrationTokens(ctx context.Context, tx *sql.Tx, returnAll bool, valid bool) ([]api.RegistrationToken, error) {
+	var stmt *sql.Stmt
+	var tokens []api.RegistrationToken
+	var tokenString sql.NullString
+	var pending, completed, usesAllowed sql.NullInt32
+	var expiryTime sql.NullInt64
+	var rows *sql.Rows
+	var err error
+	if returnAll {
+		stmt = s.listAllTokensStatement
+		rows, err = stmt.QueryContext(ctx)
+	} else if valid {
+		stmt = s.listValidTokensStatement
+		rows, err = stmt.QueryContext(ctx, time.Now().UnixNano()/int64(time.Millisecond))
+	} else {
+		stmt = s.listInvalidTokenStatement
+		rows, err = stmt.QueryContext(ctx, time.Now().UnixNano()/int64(time.Millisecond))
+	}
+	if err != nil {
+		return tokens, err
+	}
+	for rows.Next() {
+		err = rows.Scan(&tokenString, &pending, &completed, &usesAllowed, &expiryTime)
+		if err != nil {
+			return tokens, err
+		}
+		tokenString := tokenString.String
+		pending := pending.Int32
+		completed := completed.Int32
+		usesAllowed := getReturnValueForInt32(usesAllowed)
+		expiryTime := getReturnValueForInt64(expiryTime)
+
+		tokenMap := api.RegistrationToken{
+			Token:       &tokenString,
+			Pending:     &pending,
+			Completed:   &completed,
+			UsesAllowed: usesAllowed,
+			ExpiryTime:  expiryTime,
+		}
+		tokens = append(tokens, tokenMap)
+	}
+	return tokens, nil
+}
+
+func getReturnValueForInt32(value sql.NullInt32) *int32 {
+	if value.Valid {
+		returnValue := value.Int32
+		return &returnValue
+	}
+	return nil
+}
+
+func getReturnValueForInt64(value sql.NullInt64) *int64 {
+	if value.Valid {
+		returnValue := value.Int64
+		return &returnValue
+	}
+	return nil
+}
+
+func (s *registrationTokenStatements) GetRegistrationToken(ctx context.Context, tx *sql.Tx, tokenString string) (*api.RegistrationToken, error) {
+	stmt := sqlutil.TxStmt(tx, s.getTokenStatement)
+	var pending, completed, usesAllowed sql.NullInt32
+	var expiryTime sql.NullInt64
+	err := stmt.QueryRowContext(ctx, tokenString).Scan(&pending, &completed, &usesAllowed, &expiryTime)
+	if err != nil {
+		return nil, err
+	}
+	token := api.RegistrationToken{
+		Token:       &tokenString,
+		Pending:     &pending.Int32,
+		Completed:   &completed.Int32,
+		UsesAllowed: getReturnValueForInt32(usesAllowed),
+		ExpiryTime:  getReturnValueForInt64(expiryTime),
+	}
+	return &token, nil
+}
+
+func (s *registrationTokenStatements) DeleteRegistrationToken(ctx context.Context, tx *sql.Tx, tokenString string) error {
+	stmt := s.deleteTokenStatement
+	res, err := stmt.ExecContext(ctx, tokenString)
+	if err != nil {
+		return err
+	}
+	count, err := res.RowsAffected()
+	if err != nil {
+		return err
+	}
+	if count == 0 {
+		return fmt.Errorf("token: %s does not exists", tokenString)
+	}
+	return nil
```

Comment:
Probably best to just ignore the result, no need to error if the statement no-ops anyway.